controllers/userController.js

```diff
@@ -157,3 +158,86 @@ exports.userLogin = async (req, res) => {
     return res.status(200).json({ message: "Logged in!", authToken: token });
   });
 };
+
+// POST recover password
+exports.userRecoverPassword = async (req, res) => {
+  // Get the data from the request
+  const { email } = req.body;
+  // Check if the user exists
+  User.findOne({ email: email }, async (err, user) => {
+    if (!user) {
+      return res.status(404).json({ error: "User not found" });
+    }
+    // Generate a temporary password
+    let tempPass = Math.random().toString(36).slice(-8);
+    // Create and assign a token
+    let token = jwt.sign({_id: user._id, pass: tempPass}, 'RECOVER_SECRET', {
+      expiresIn: '1h'
+    });    
+    // Send email with the token    
+    const transporter = nodemailer.createTransport({
+      host: 'smtp.ethereal.email',
```

Comment:
I'm having an error when sending the email, the console output is shown in the attached image
![image](https://user-images.githubusercontent.com/112887440/218335433-01583c4b-77d5-41b3-9c46-c6b15e5bf314.png)

I'm gessing it's because  it cannot connect to this email service. Can you explain where this `smtp.ethereal.email` and `rosalee.williamson29@ethereal.email` come from?

Reply:
The connection details come from this service: https://ethereal.email/create